src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/Handle.cs

```diff
@@ -159,7 +160,7 @@ internal int Token
             }
         }
 
-        public override bool Equals(object? obj)
+        public override bool Equals([NotNullWhen(true)] object? obj)
         {
             return obj is Handle && Equals((Handle)obj);
```

Comment:
Please remove extra cast, please do same all below occurrences, not gonna comment more